a3p-integration/proposals/z:acceptance/auction.test.js

```diff
@@ -0,0 +1,389 @@
+/**
+ * @file In this file we aim to test auctioneer in an isolated manner. Here's the scenario to test;
+ * - Send 100 ATOMs to gov1 from validator
+ * - For book0, ATOM is collateral, set two types of bids; by price and by percentage, user1 is the bidder
+ * - Deposit 100 ATOMs into book0, gov1 is the depositor
+ * - Wait until placed bids get their payouts
+ * - Make sure the depositor gets correct amounts
+ */
+
+/** @typedef {import('./test-lib/sync-tools.js').RetyrOptions} RetryOptions */
+
+import {
+  addPreexistingOracles,
+  agd,
+  agopsInter,
+  agoric,
+  ATOM_DENOM,
+  CHAINID,
+  executeOffer,
+  getPriceQuote,
+  getUser,
+  GOV1ADDR,
+  GOV3ADDR,
+  pushPrices,
+  USER1ADDR,
+  VALIDATORADDR,
+} from '@agoric/synthetic-chain';
+import '@endo/init';
+import test from 'ava';
+import { boardSlottingMarshaller, makeFromBoard } from './test-lib/rpc.js';
+import {
+  retryUntilCondition,
+  waitUntilAccountFunded,
+  waitUntilOfferResult,
+} from './test-lib/sync-tools.js';
+import { AmountMath } from '@agoric/ertp';
+
+export const scale6 = x => BigInt(Math.round(x * 1_000_000));
```

Comment:
I think I prefer this because it warns if the caller specified more digits than will end up in the result. Do we ever use this with calculated Numbers rather than manifest constants?

```suggestion
export const scale6 = x => BigInt(x * 1_000_000);
```

Reply:
No, we use it with only constants from `config`